core/model/modx/processors/resource/create.class.php

```diff
@@ -633,6 +638,26 @@ public function checkIfSiteStart() {
         return $saved;
     }
 
+    /**
+     * Check that the symlink target is a valid resource ID
+     * @return bool
+     */
+    public function checkSymLinkTarget() {
+        $target = $this->getProperty('content', null);
+
+        if ($target === null || $target === '') {
+            return true;
+        }
+
+        $targetResource = $this->modx->getObject('modResource', $target);
```

Comment:
Make this more strict and require the ID to be an integer:

```suggestion
        $targetResource = $this->modx->getObject('modResource', array('id' => (int)$target));
```

Reply:
Sounds like a good idea. I'll do that, I have just not gotten the time to look more at this yet.

Reply:
If you just cast it into an integer, i.e. a value of `123abc` would try to retrieve the resource 123. And if that resource exists, the value is valid. So it would be better to check it with `filter_var` in my opinion.

Reply:
I have added the integer check.